photomanagement/photomanagement/database.py

```diff
@@ -67,18 +67,41 @@ def __init__(self, path: pathlib.Path | None) -> None:
         if not self.image_directory_path.exists():
             self.image_directory_path.mkdir()
 
-    def add_images_from_directory(self, photo_dir: pathlib.Path):
+    def add_images_from_directory(self, photo_dir: pathlib.Path) -> list[Photo]:
+        '''
+        Adds all images from a directory to the database.
+        Also generates any required information. 
+
+        Throws IOError if a filepath does not lead to an image.
+        All images before the IOError will be inserted into the database
+        Returns a list of the photos.
```

Comment:
Is this desired behavior? It won't continue looping through the directory, it'll stop at the first file that's not an image. 

We should just skip over that file imo